packages/components/src/ClickableStyle/ClickableStyle.tsx

```diff
@@ -23,6 +23,33 @@ export type ClickableStyleProps<IComponent extends React.ElementType> = {
    * The style of the element.
    */
   variant: "flat" | "outline" | "link";
+  /**
+   * Optional icon that can be positioned to the left or right of the other button contents.
+   *
+   * Default position is 'left'. Use the iconPlacement prop to move it to the right side.
+   */
+  icon?: {
```

Comment:
(You may have already answered this elsewhere)

Why do we need to add this, when we can already pass in an icon as `children`?

```jsx
<Button>
  <FooIcon>
  Hello!
</Button>
```

Is it because of the height/sizing issue mentioned in the PR description?

Is it definitely worth the added props (that will inevitably end up being insufficient for what someone needs to do)?

Reply:
they can, but there's some consistent styling that's needed every time. for example, this is what it looks like to have an icon in children:
<img width="51" alt="Screen Shot 2021-09-21 at 11 13 29 AM" src="https://user-images.githubusercontent.com/15840841/134225063-dbd764c1-1788-4d5f-bcf0-b6435a28ba90.png">

Diedra & I discussed that when we add tailwind utility classes to traject, this would provide less value because you could easily just add this bit of styling through the className. But for now, I think it's worth us making the most common case easier.

Reply:
One thing to consider with making a separate component is devs will need to pass in the element again. I'm also not sure what we would call it since it could produce either a button or an anchor element (or a react router link; who knows). At this point, I'm mostly concerned that making a separate component would be confusing.

Reply:
For why we're adding this at all and not asking people to pass in the icon as a child, the only added value (I think) is the spacing consistency.

The spacing argument is harder to make because of the square viewbox issue – I don't know what kind of spacing variance we would see if devs add their own spacing vs using this component and possibly adjusting for the extra space (but probably not). Right now, I actually feel like the benefit of the `iconHorizontalMargin ` is that the component API calls out the issue and suggests a solution. The solution itself (adding negative margin via a style attribute) is not interesting, but I think people are more likely to even notice there's a problem and see the solution because of this prop.

We're also setting the size and color of the svg, but that can be set on the button styles even if we don't know for sure if there's an icon in there.

Reply:
Oh, I just thought of another benefit, which is that we can discourage people from using the "plain" (formerly "text") variant without an icon. They can get around it by just passing the icon in via the children as usual, so you can still get around our little fence if you want, but at least we're making people go around.

Reply:
My argument against adding this is – is it annoying for people to learn this API? I don't think this is a common pattern, so people will be learning it from scratch and reminding each other in PRs to use it instead of passing the icon in via the children. Are the benefits of this change worth that time and effort?

Reply:
Well I agree with everything that's been said 😂  But most of the benefits are imo not dealbreakers:
 - as long as designers follow our guidance for the "plain" variant, it should be fine not checking that in the code
 - spacing variance from devs styling it themselves doesn't bother me so much, unless it's bothering designers

I'm starting to waver on how much benefit the spacing consistency gives over the cost of a more complicated API. Is it possible to land the smaller changes (setting size & color of svg on the button styles) independent of the API change? I'd like to think it through some more / maybe get the opinion of a product dev (so we're not inventing pains that don't exist). and if it's easier we could all hop in a discussion circle haha

Reply:
Always happy to hop in a discussion circle.

There's probably no 100% right or wrong answer here. I'm mostly thinking of component API and what kinds of things do and don't belong in a component.

But you're both closer to the actual work/needs than I am. We can always go with what meets those needs (always a good place to start 😸) and iterate/learn from there.

Reply:
I'm totally fine with just landing the css changes so icons assume the expected height and color. I'll do that in a separate PR so we can easily reuse this one if we for some reason change our minds later.

I'll also add some spacing to the existing icon button stories and add stories for just an icon so people can see from storybook how we are expecting icons to be used in buttons.

Is there further guidance we should leave for people who may be expecting an icon button component/variant? Probably in zeroheight?